hadoop-crypto/src/test/java/com/palantir/crypto2/hadoop/EncryptedFileSystemTest.java

```diff
@@ -256,6 +256,18 @@ public void testRename_failedRename() throws IOException {
         verify(mockKeyStore).remove(newPath.toString());
     }
 
+    @Test
+    public void testRename_failedRenameBecauseDestinationFileAlreadyExists() throws IOException {
+        when(mockFs.exists(newPath)).thenReturn(true);
+        when(mockFs.rename(path, newPath)).thenReturn(false);
+
+        boolean renamed = mockedEfs.rename(path, newPath);
+
+        assertThat(renamed).isFalse();
+        verify(mockKeyStore, never()).remove(path.toString());
+        verify(mockKeyStore, never()).remove(newPath.toString());
```

Comment:
I think if you added `verify(mockKeyStore, never()).put(newPath.toString(), any());` you'd end up getting a failure